docs/actions.md

```diff
@@ -0,0 +1,63 @@
+# BpyBuild Actions
+
+One feature of BpyBuild is the ability to write scripts and run them at build time. This page goes over how to write a build script.
+
+# Quick Start
+Here's an example of a simple action:
+```yaml
+# bpy-build.yaml
+build_actions:
+  dev: 
+    script: "dev.py"
+```
+
+```py
+# dev.py
+def main() -> None:
+    with open("mcprep_dev.txt", "w") as f:
+        f.write("hi guys c:")
+
+# Not required, but useful for testing
+if __name__ == "__main__":
+    main()
+```
+
+> [!CAUTION]
+> **Never** put action related code in the global scope. 
+>
+> BpyBuild doesn't run scripts in the traditional sense. Instead, BpyBuild imports scripts as Python modules and runs `main`. As such, global code is executed on import, which is before the actual build process.
+
+# Hooks
+BpyBuild also supports the concept of hooks. Currently, the following hooks are supported:
+- `main`: executed during the build process; working directory is set to a copy of the source tree under `build/stage-1`
+- `pre_build`: executed before building; working directory is set to the `addon_folder` variable defined in `bpy-build.yaml`
+- `pre_install`: executed before installing the built addon; working directory is set to `build/`
+- `post_install`: executed after installing the built addon; working directory is set to the addons folder of the Blender version last installed to
```

Comment:
I mention below the request to get in the path of the install location, but I suppose it could also work if as described above. However, I'm not finding this to be the case in practice, with this code:

```python
def post_install() -> None:
    print("POST INSTALL")
    print(os.getcwd())
```

I'm getting this print out (ie my github folder loc)

```
POST INSTALL
/Users/patrickcrawford/Documents/gits/mcprep
```

Reply:
Works fine on my end:
```
Installed to /home/mahid/.config/blender/3.5/scripts/addons
POST INSTALL
/home/mahid/.config/blender/3.5/scripts/addons
Installed to /home/mahid/.config/blender/3.4/scripts/addons
POST INSTALL
/home/mahid/.config/blender/3.4/scripts/addons
Installed to /home/mahid/.config/blender/2.80/scripts/addons
POST INSTALL
/home/mahid/.config/blender/2.80/scripts/addons
```